server/src/main/java/io/druid/metadata/IndexerSQLMetadataStorageCoordinator.java

```diff
@@ -85,27 +89,48 @@ public void start()
     connector.createSegmentTable();
   }
 
+  @Override
   public List<DataSegment> getUsedSegmentsForInterval(
       final String dataSource,
       final Interval interval
   ) throws IOException
   {
+    return getUsedSegmentsForIntervals(dataSource, ImmutableList.of(interval));
+  }
+
+  @Override
+  public List<DataSegment> getUsedSegmentsForIntervals(
```

Comment:
good catch.
but, it wouldn't matter because list of intervals is always list of "disjoint" intervals ensured by calling JodaUtils.condenseIntervals(..). In case of 2 intervals overlapping same segment, this would have given same segment twice, which again wouldn't matter because caller uses "windowed" segments appropriately.
however, that looks weird from api perspective so updated the code to remove duplicates and also updated the test case to verify same.